lg_earth/python_scripts/toggle_layer.py

```diff
@@ -0,0 +1,241 @@
+    #     if color_state in ['FFFFFF']:
+    #         return 1
+    #     elif color_state in ['000000', '2F2F2F', '9A9A9A', 'BCB9B6', 'F2F2F2', 'F4F4F4']:
+    #         return 0
+    #     else:
+    #         print(layer_y, " !!!!!!!!!color_state: ", color_state)
+    #         return None
+
+    def get_state(self, layer_y):  # TODO could work for any but it is set for buildings rn
+        if self.states_image is None:
+            try:
+                root = self.display.screen().root
+                adjusted_x = self.offset + self.checks_column
+                self.states_image = root.get_image(adjusted_x, 0, 1, 1920, X.ZPixmap, 0xffffffff)
+            except Exception as e:
+                print(e)
+                return None
+        pixel_value_bytes = self.states_image.data[layer_y * self.stride: (layer_y + 1) * self.stride]
+        pixel_value = int.from_bytes(pixel_value_bytes, byteorder='little')
+        color_state = hex(pixel_value)[2:].zfill(6).upper()
+        print("layer_row: ", layer_y, " hex_state: ", color_state)
+        if color_state == 'FFFFFF':
+            return 1
+        elif color_state in ['000000', '2F2F2F', '9A9A9A', 'BCB9B6', 'F2F2F2', 'F4F4F4']:
+            return 0
+        else:
+            return None
+
+def find_and_create_earths():
+    try:
+        window_ids = subprocess.check_output(['xdotool', 'search', '--onlyvisible', '--name', 'Google Earth']).decode().strip().split()
+    except Exception as e:
+        print("Error: 'xdotool' failed searching for earths:\n", e)
+        return []
+
+    EARTHS = []  # Earth objects list
+    for win_id in window_ids:
+        try:
+            geometry_info = subprocess.check_output(['xdotool', 'getwindowgeometry', '--shell', win_id]).decode().strip()
+            offset = int(geometry_info.split('X=')[1].split()[0])
+            EARTHS.append(Earth(win_id, offset))
+        except Exception as e:
+            print(f"Failed to retrieve window geometry for window ID: {win_id}:\n", e)
+    sorted_earths = sorted(EARTHS, key=lambda earth: earth.offset)  # Sort by offsets
+    center_index = len(sorted_earths) // 2
+    if len(sorted_earths) in [3, 5, 7]:  #mark center and change order of toggle
+        sorted_earths[center_index].is_master = True
+        newsort_earths = [sorted_earths[center_index]] + [x for pair in zip(sorted_earths[center_index-1::-1], sorted_earths[center_index+1:]) for x in pair]
+        print("earth xdotool ids from center out on canvas:", *[earth.win_id for earth in newsort_earths])
+        return newsort_earths
+    else:
+        sorted_earths[0].is_master = True
+        print("earth xdotool ids left to right on canvas:", *[earth.win_id for earth in sorted_earths])
+        return sorted_earths
+
+
+if __name__ == "__main__":
+    onoff = 2  #default to toggle
+    layer = "buildings" #default to buildings layer
+# handle command line arguments
+    if len(sys.argv) == 2:
```

Comment:
`argparse` is always a bit more fun to play with than `sys.argv`. https://github.com/EndPointCorp/lg_ros_nodes/blob/814e672ba30113b86edb26c42fc6e86a98fae3e4/spacenav_remote/scripts/client.py#L5 is a good example of using it. Not needed for this, but a good thing to look at in the future.